pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -977,12 +1008,15 @@ func verifyStatus(ctx *domainContext, status *types.DomainStatus) {
 				status.SetErrorDescription(errDescription)
 			}
 
-			//cleanup app instance tasks
-			if err := hyper.Task(status).Delete(status.DomainName); err != nil {
-				log.Errorf("failed to delete domain: %s (%v)", status.DomainName, err)
-			}
-			if err := hyper.Task(status).Cleanup(status.DomainName); err != nil {
-				log.Errorf("failed to cleanup domain: %s (%v)", status.DomainName, err)
+			// in cluster mode, we can not delete the pod due to failing to get app info
```

Comment:
Was the issue this is fixing something appearing when there is a failover/takeover and another node in the cluster starts running the app instance?
Or is it something which could happen when an app instance is first provisioned on the cluster?

Reply:
This can happen even on the first node of the app deployment, sometimes we can not get the status from the k3s cluster, or somethings it takes time to come up running state, but we should not remove this kubernetes configuration, it has the config stored in the database, it has it's own scheduling and control process to eventually bring it to the intended state. If we delete the config from the cluster, then we need to wait for another 10 minutes to retry, etc. and it will cause confusing.

Reply:
It would be good to capture the above explanation either in a comment here or in pkg/pillar/docs/zedkube.md

Reply:
Sure. I documented this in zedkube.md, and referenced from domainmgr.go

Reply:
Hmm - re-reading it and it still looks odd.
Does the kubevirt Info() return an error or does it return HALTED when the issue is merely that it can't (yet) fetch the status from k3s? 
Can we not fix that Info() to return something more accurate?

If it returns an error or HALTED then this function will set an error, and that error might be propagated to the controller, which would be confusing if the task is slow at starting, or is starting on some other node.

So I think this check is in the wrong place.

Reply:
current code in kubevirt.go logic is that
  1) if not found this app, then it will return status "", and  an error logError("getVMIStatus: No VMI %s found with the given nodeName %s", repVmiName, nodeName)
  2) if found this app on another node in cluster, then it returns status "nolocal", no error
  3) if found on this node, then return whatever the kubernetes app running status

  with above condition, if error is returned, then status is set to types.Broken
  we further map the above status in string with a mapping:

```
// Use few states  for now
var stateMap = map[string]types.SwState{
    "Paused":     types.PAUSED,
    "Running":    types.RUNNING,
    "NonLocal":   types.RUNNING,
    "shutdown":   types.HALTING,
    "suspended":  types.PAUSED,
    "Pending":    types.PENDING,
    "Scheduling": types.SCHEDULING,
    "Failed":     types.FAILED,
}
```
this is a good point of the error condition if not running will be confusing. I can change the condition 1) above to 'Scheduling', and it's a currently defined state, and sort of reflecting the kubernetes app status.